java/spi.java.hints/src/org/netbeans/modules/java/hints/spiimpl/Utilities.java

```diff
@@ -1090,7 +1091,7 @@ public synchronized ClasspathInfo createUniversalCPInfo() {
                     .filter((p) -> "j2se".equals(p.getSpecification().getName()))
                     .filter((p) -> p.getSpecification().getVersion() != null)
                     .filter((p) -> p.getSpecification().getVersion().compareTo(maxVersion) < 0)
-                    .max((p1, p2) -> p1.getSpecification().getVersion().compareTo(p2.getSpecification().getVersion()))
+                    .max(Comparator.comparing((p) -> p.getSpecification().getVersion()))
```

Comment:
to do this I had add generics to `SpecificationVersion`'s `Comparable` interface.

i believe this should be safe since the class is final and the `compareTo` method would have thrown a `ClassCastException` anyway when anything other than SpecificationVersion would be passed as parameter.

This requires a sig file change. 

@neilcsmith-net could you take a look at the second commit if that is ok? The second commit is entirely optional, no big deal to drop it. I didn't bump any versions.

Reply:
@mbien I _think_ that's fine.  And leads to nicer code here.  Still binary compatible because of bridge method not picked up by sigtest?  @jtulach the best person to check with on sigtests and compatible API updates.